
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
        <link rel="prev" href="../profile/">
      
      
        <link rel="next" href="../outputs/">
      
      
      <link rel="icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.5.3, mkdocs-material-9.5.7">
    
    
      
        <title>Parameters - TOSTADAS</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.f2e4d321.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    <body dir="ltr">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="TOSTADAS" class="md-header__button md-logo" aria-label="TOSTADAS" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            TOSTADAS
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Parameters
            
          </span>
        </div>
      </div>
    </div>
    
    
      <script>var media,input,key,value,palette=__md_get("__palette");if(palette&&palette.color){"(prefers-color-scheme)"===palette.color.media&&(media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']"),palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent"));for([key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="TOSTADAS" class="md-nav__button md-logo" aria-label="TOSTADAS" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    TOSTADAS
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Home
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../installation/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Installation
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../quickstart/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Quick Start
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../profile/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Profile Options and Input Files
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  <span class="md-ellipsis">
    Parameters
  </span>
  

      </a>
      
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../outputs/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Outputs
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../submission/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Submission Guide
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../links/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Helpful Links
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../getintouch/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Get in Touch
  </span>
  

      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


  <h1>Parameters</h1>

<p>Parameters
RamiyapriyaS edited this page 5 hours ago · 3 revisions
Default parameters are given in the nextflow.config file. This table lists the parameters that can be changed to a value, path or true/false. When changing these parameters pay attention to the required inputs and make sure that paths line-up and values are within range. To change a parameter you may change with a flag after the nextflow command or change them within your standard_params.config or standard.yaml file.</p>
<p>Please note the correct formatting and the default calculation of submission_wait_time at the bottom of the params table.
7.1 Input Files
Param   Description Input Required
--fasta_path    Path to directory containing single sample fasta files  Yes (path as string)
--ref_fasta_path    Reference Sequence file path    Yes (path as string)
--meta_path Meta-data file path for samples Yes (path as string)
--ref_gff_path  Reference gff file path for annotation  Yes (path as string)
--bakta_db_path Path to Bakta reference database    Yes (path as string)
--env_yml   Path to environment.yml file    Yes (path as string)
7.2 Run Environment
Param   Description Input Required
--scicomp   Flag for whether running on Scicomp or not  Yes (true/false as bool)
--docker_container  Name of the Docker container    Yes, if running with docker profile (name as string)
--docker_container_vadr Name of the Docker container to run VADR annotation Yes, if running with docker profile (name as string)
--docker_container_bakta    Name of the Docker container to run Bakta annotation    Yes, if running with docker profile (name as string)
7.3 General Subworkflow
Param   Description Input Required
--run_submission    Toggle for running submission   Yes (true/false as bool)
--run_liftoff   Toggle for running liftoff annotation   Yes (true/false as bool)
--run_vadr  Toggle for running vadr annotation  Yes (true/false as bool)
--run_bakta Toggle for running Bakta annotation Yes (true/false as bool)
--cleanup   Toggle for running cleanup subworkflows Yes (true/false as bool)
7.4 Cleanup Subworkflow
Param   Description Input Required
--clear_nextflow_log    Clears nextflow work log    Yes (true/false as bool)
--clear_nextflow_dir    Clears nextflow working directory   Yes (true/false as bool)
--clear_work_dir    Param to clear work directory created during workflow   Yes (true/false as bool)
--clear_conda_env   Clears conda environment    Yes (true/false as bool)
--clear_nf_results  Remove results from nextflow outputs    Yes (true/false as bool)
7.5 General Output
Param   Description Input Required
--output_dir    File path to submit outputs from pipeline   Yes (path as string)
--overwrite_output  Toggle to overwriting output files in directory Yes (true/false as bool)
7.6 Metadata Validation
Param   Description Input Required
--val_output_dir    File path for outputs specific to validate sub-workflow Yes (folder name as string)
--val_date_format_flag  Flag to change date output  Yes (-s, -o, or -v as string)
--val_keep_pi   Flag to keep personal identifying info, if provided otherwise it will return an error   Yes (true/false as bool)
--validate_custom_fields    Toggle checks/transformations for custom metadata fields on/off No (true/false as bool)
--custom_fields_file    Path to the JSON file containing custom metadata fields and their information   No (path as string)
7.7 Liftoff
Param   Description Input Required
--final_liftoff_output_dir  File path to liftoff specific sub-workflow outputs  Yes (folder name as string)
--lift_print_version_exit   Print version and exit the program  Yes (true/false as bool)
--lift_print_help_exit  Print help and exit the program Yes (true/false as bool)
--lift_parallel_processes   # of parallel processes to use for liftoff  Yes (integer)
--lift_delete_temp_files    Deletes the temporary files after finishing transfer    Yes (true/false as string)
--lift_child_feature_align_threshold    Only if its child features usually exons/CDS align with sequence identity â‰¥S  designate a feature mapped
--lift_unmapped_feature_file_name   Name of unmapped features file name Yes (path as string)
--lift_copy_threshold   Minimum sequence identity in exons/CDS for which a gene is considered a copy; must be greater than -s; default is 1.0   Yes (float)
--lift_distance_scaling_factor  Distance scaling factor; by default D =2.0  Yes (float)
--lift_flank    Amount of flanking sequence to align as a fraction of gene length   Yes (float between [0.0-1.0])
--lift_overlap  Maximum fraction of overlap allowed by 2 features   Yes (float between [0.0-1.0])
--lift_mismatch Mismatch penalty in exons when finding best mapping; by default M=2 Yes (integer)
--lift_gap_open Gap open penalty in exons when finding best mapping; by default GO=2    Yes (integer)
--lift_gap_extend   Gap extend penalty in exons when finding best mapping; by default GE=1  Yes (integer)
--lift_infer_transcripts    Use if annotation file only includes exon/CDS features and does not include transcripts/mRNA    Yes (True/False as string)
--lift_copies   Look for extra gene copies in the target genome Yes (True/False as string)
--lift_minimap_path Path to minimap if you did not use conda or pip Yes (N/A or path as string)
--lift_feature_database_name    Name of the feature database, if none, then will use ref gff path to construct one  Yes (N/A or name as string)
--repeat_lib    Path to repeatmasker library    No, default is mpox
7.8 VADR
Param   Description Input Required
--vadr_output_dir   File path to vadr specific sub-workflow outputs Yes (folder name as string)
--vadr_models_dir   File path to models for MPXV used by VADR annotation    Yes (folder name as string)
7.9 Bakta
Param   Description Input Required
--bakta_db_path Path to Bakta database if user is supplying database    No (path to database)
--download_bakta_db Option to download Bakta database   Yes (true/false)
--bakta_db_type Bakta database type (light or full) Yes (string)
--bakta_output_dir  File path to bakta specific sub-workflow outputs    Yes (folder name as string)
--bakta_min_contig_length   Minimum contig size Yes (integer)
--bakta_threads Number of threads to use while running annotation   Yes (integer)
--bakta_genus   Organism genus name Yes (N/A or name as string)
--bakta_species Organism species name   Yes (N/A or name as string)
--bakta_strain  Organism strain name    Yes (N/A or name as string)
--bakta_plasmid Name of plasmid Yes (unnamed or name as string)
--bakta_locus   Locus prefix    Yes (contig or name as string)
--bakta_locus_tag   Locus tag prefix    Yes (autogenerated or name as string)
--bakta_translation_table   Translation table   Yes (integer)
--bakta_db_path Path to Bakta reference database    Yes (path as string)
--bakta_db_type Type of database to use for annotation  Yes (full/light as bool)
--download_bakta_db Toggle to download Bakta database   Yes (true/false as bool)
--run_bakta Toggle for running Bakta annotation Yes (true/false as bool)
7.10 Sample Submission
Param   Description Input Required
--submission_output_dir Either name or relative/absolute path for the outputs from submission   Yes (name or path as string)
--submission_prod_or_test   Whether to submit samples for test or actual production Yes (prod or test as string)
--submission_config Configuration file for submission to public repos   Yes (path as string)
--submission_wait_time  Calculated based on sample number (3 * 60 secs * sample_num)    integer (seconds)
--batch_name    Name of the batch to prefix samples with during submission  Yes (name as string)
--send_submission_email Toggle email notification on/off    Yes (true/false as bool)
--req_col_config    Path to the required_columns.yaml file  Yes (path as string)
--processed_samples Path to the directory containing processed samples for update only submission entrypoint (containing <batch_name>.<sample_name> dirs)   Yes (path as string)
❗ Important note about send_submission_email: An email is only triggered if Genbank is being submitted to AND table2asn is the genbank_submission_type. As for the recipient, this must be specified within your submission config file under 'general' as 'notif_email_recipient'*</p>
<p>7.11 Entrypoint and User Provided Annotation
Param   Description Input Required
--final_split_metas_path    Full path directly to the directory containing validated metadata file(s) (1 per sample)    Yes (path as string)
--final_annotated_files_path    Full path directly to the directory containing annotation file(s) (1 per sample)    Yes (path as string)
-- final_split_fastas_path  Full path directly to the directory containing fasta(s) (1 per sample)  Yes (path as string)
Pages 10
Home
Overview
Pipeline Summary
Installation
Environment Setup
Quick Start
Validate Environment
Pipeline Execution Examples
Profile Options and Input Files
Parameters
Outputs
Submission Guide
Helpful Links
Get In Touch
Clone this wiki locally</p>












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    <script id="__config" type="application/json">{"base": "..", "features": [], "search": "../assets/javascripts/workers/search.b8dbb3d2.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../assets/javascripts/bundle.caa56a14.min.js"></script>
      
    
  </body>
</html>